### ISSUE TITLE
Crash on DRB171 (intraConstantProp)

### ISSUE THREAD

**Issue by BradSwain** [coderrect-inc/OpenRace#237]
**Describe the bug**
The tool crashes in debug mode on DRB171 due to a failed assertion.

```shell
llvm/lib/IR/Value.cpp:403: void llvm::Value::doRAUW(llvm::Value*, llvm::Value::ReplaceMetadataUses): Assertion `New->getType() == getType() && "replaceAllUses of value with new value of different type!"' failed.
```

The stack dump indicates that the crash happens during the intrConstProp function:

https://github.com/coderrect-inc/OpenRace/blob/829e28a7d2fc4d760c56b079423d12ae5b176879/src/PreProcessing/Passes/OMPConstantPropPass.cpp#L110

**Expected behavior**
Should not crash.

**To Reproduce**
Run a debug build of the tool against `tests/data/integration/dataracebench/DRB171-threadprivate3-orig-no.ll`

**Comment by BradSwain** [comment]
The crash is fixed but report is wrong.

I will open a new issue after #245 is merged.
